signingworker/utils.py

```diff
@@ -0,0 +1,32 @@
+import fcntl
+import socket
+import struct
+import hashlib
+import functools
+
+
+def get_ip_address(interface_name):
+    """Return IP address used by specific interface
+
+    :rtype : str
+    :param interface_name: Interface name, e.g. eth0
+    :return: default IP address used by interface
+    """
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    return socket.inet_ntoa(fcntl.ioctl(
+        s.fileno(),
+        0x8915,  # SIOCGIFADDR
+        struct.pack('256s', interface_name[:15])
+    )[20:24])
+
+
+# TODO: find a better way to specify source IP, maybe change the signing API
+my_ip = get_ip_address("eth0")
+
+
+def get_hash(path, hash_type="sha512"):
+    h = hashlib.new(hash_type)
+    with open(path, "rb") as f:
```

Comment:
I wish there was something in stdlib for this common pattern. :(
I'm going to leave it as is for two reasons:
1) it works :)
2) it's used in multiple places, so it's at least 1 LOC less this way :)